camayoc/tests/remote/rho/test_scan.py

```diff
@@ -90,42 +86,24 @@ def test_scan(isolated_filesystem, profile):
     assert rho_scan.exitstatus == 0, logfile
     assert os.path.isfile(reportfile)
 
-    # we will collect errors in scanned facts and report at end of test on the
-    # results if collected facts do not match expected values.
-    scan_errors = []
-    with open(reportfile) as csvfile:
-        scan_results = csv.DictReader(csvfile)
-        # each row corresponds to a scanned host
-        for row in scan_results:
-            known_facts = {}
-            for host in cfg['rho']['hosts']:
-                # find the facts for the scanned host we are inspecting
-                if host['ip'] == row.get('connection.host'):
-                    known_facts = host['facts']
-                    break
-            missed_facts = known_facts.keys() - row.keys()
-            for fact in missed_facts:
-                msg = ('Expected {} fact in scan of {} host,'
-                       ' but it is missing'.format(fact, host['ip']))
-                scan_errors.append(msg)
-            # now we will test all facts that are indeed present
-            for fact in (known_facts.keys() - missed_facts):
-                if fact == 'cpu.bogomips':
-                    # this is a transitory measurement that is likely to
-                    # change from scan to scan
-                    continue
-                try:
-                    assert (str(known_facts[fact]) == str(row.get(fact)))
-                except AssertionError:
-                    msg = ('Test failed on host {} in profile {}.'
-                           ' \n\tScan found {} = {} instead of {}.'.format(
-                            host['ip'],
-                            profile['name'],
-                            fact,
-                            row.get(fact),
-                            known_facts[fact],
-                           ))
-                    scan_errors.append(msg)
-    if scan_errors:
-        msg = '\n'.join(e for e in scan_errors)
-        raise AssertionError(msg)
+
+@pytest.mark.facts_needed
+@pytest.mark.parametrize('fact', profiles())
+def test_facts(fact):
+    """Test each fact from the scan reports.
+
+    :id: d9eb29bd-1b61-421a-b680-f494e868b11e
+    :description: Test the facts from a scan report
+    :steps:
+            1) Test facts collected from scans against expected data
```

Comment:
I usually indent the line using 4 spaces, please check if this is being rendered on the docs as a numbered list. It should but I think it worth double checking.